lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -994,6 +978,10 @@ DotOperandEncodingAttr::getElemsPerThread(ArrayRef<int64_t> shape,
 unsigned DotOperandEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
                                                         Type eltTy) const {
   if (auto mmaParent = mlir::dyn_cast<MmaEncodingTrait>(getParent())) {
+    if (auto nvidiaMmaParent = mlir::dyn_cast<NvidiaMmaEncodingAttr>(mmaParent);
+        nvidiaMmaParent && nvidiaMmaParent.isAmpere()) {
+      return product<unsigned>(getElemsPerThread(shape, eltTy));
```

Comment:
Any chance we can leave this as the implementation fo this function and completely kill `getTotalElemsPerThreadForOperand` and the logic below? If it's too much effort, let's just leave a TODO comment.

Reply:
Just looked around and the answer is no, it's not easy.

Once we completely kill Volta's logic we'll be able to improve the state of things tho.